### PR TITLE
Feat/auth

### PR DESCRIPTION
**Pull request by purvasharmadev**
### Added Auth
- login, logout
- SignIn
- Guest login
- Private Routes
- Restricted Routes

### Files to Check

[Click Here](https://github.com/purvasharmadev/kvideo/pull/6/files/35db2f7d2d73f1d27c4733ac7494cc4c27326461..9c84b33b53bb1dd461d24300922f270ced318f65)

### CODE REVIEW

**Review by adityapandadev — APPROVED** [review]

File: src/Auth/Login/login.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        email: userData.email,
+        password: userData.password,
+      },
+    });
+  };
+
+  const { response, error, fetchData } = useAxios();
+
+  useEffect(() => {
+    if (response !== undefined && response.foundUser) {
```
Comment (itsmebipul00) [inline src/Auth/Login/login.js:42]: Suggestion: reponse?.foundUser , optional chaning could be used

File: src/Auth/Signup/signup.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        email: userData.email,
+        password: userData.password,
+      },
+    });
+  };
+
+  const { response, error, fetchData } = useAxios();
+
+  useEffect(() => {
+    if (response !== undefined && response.createdUser) {
```
Comment (itsmebipul00) [inline src/Auth/Signup/signup.js:35]: Here too

File: src/Auth/Signup/signup.js [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+      toast.success("User Created Succesfully!", {
+        id: "signup-success",
+        position: toast.POSITION.TOP_RIGHT,
+        autoClose: 2000,
+      });
+      location.state !== null
+        ? navigate(location.state?.from?.pathname, { replace: true })
+        : navigate("/login", { replace: true });
+    } else {
+      console.log("error ", error);
```
Comment (itsmebipul00) [inline src/Auth/Signup/signup.js:45]: Suggestion: console log could be removed, error could be handled

File: src/router.js [hunk 4]
```diff
@@ -1,19 +1,47 @@
...
+import { RestrictedRoute } from "./Auth/AuthRoutes/RestrictedRoutes";
+
 // pages
 import { HomePage } from "./Pages/HomePage/home-page";
-import { Explore } from "./Pages/Explore/explore-page";
+import { VideoDetail } from "./Pages/Video/video-detail";
 import PageNotFound from "./Pages/PageNotFound/page-not-found";
+import { Explore } from "./Pages/Explore/explore-page";
+import { Login } from "./Auth/Login/login";
+import { Signin } from "./Auth/Signup/signup";
```
Comment (itsmebipul00) [inline src/router.js:18]: Suggestion: index file could be created to export all pages

**Review by itsmebipul00 — APPROVED** [review]
Awesome work! Good to merge

**Comment by purvasharmadev** [comment]
> Awesome work! Good to merge

Thankyou for your suggestions.

**Pull request merged**